plugins/modules/repair.py

```diff
@@ -0,0 +1,119 @@
+#!/usr/bin/python
+
+# copyright (c) 2020, Matthias Dellweg
+# GNU General Public License v3.0+ (see LICENSE or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+
+DOCUMENTATION = r"""
+---
+module: repair
+short_description: Repair corrupted artifacts in a repository version on a pulp server
+description:
+  - "Performes a repair task that identifies corrupted artifact files and attempts to refetch them."
+options:
+  repository:
+    description:
+      - Name of the repository to have a version repaired
+    type: str
+    required: true
+  version:
+    description:
+      - Version number to be repaired
+    type: int
+    required: false
+extends_documentation_fragment:
+  - pulp.squeezer.pulp.glue
+  - pulp.squeezer.pulp
+author:
+  - Matthias Dellweg (@mdellweg)
+"""
+
+EXAMPLES = r"""
+- name: Repair latest version of a repository
+  repair:
+    api_url: localhost:24817
+    username: admin
+    password: password
+    repository: my_file_repo
+- name: Repair first version of a repository
+  repair:
+    api_url: localhost:24817
+    username: admin
+    password: password
+    repository: my_file_repo
+    version: 1
+"""
+
+RETURN = r"""
+  corrupted:
+    description: Number of identified corrupted artifacts
+    type: int
+    returned: if available
+  missing:
+    description: Number of identified missing artifacts
+    type: int
+    returned: if available
+  repaired:
+    description: Number of successfully repaired artifacts
+    type: int
+    returned: if available
+"""
+
+import re
+import traceback
+
+from ansible_collections.pulp.squeezer.plugins.module_utils.pulp_glue import PulpAnsibleModule
+
+try:
+    import pulp_glue.file.context
```

Comment:
We are getting closer, but we still need to find a way to find all pulp_glue plugin context modules, it seems.
Funny that this very issue (needing plugin specifics for something theoretically generic) was the original reason I was unhappy with the original design of this module.